packages/types/src/types/models-extra/report.ts

```diff
@@ -87,9 +90,26 @@ export interface ChartData extends Record<string, unknown> {
   timestamp?: number;
 }
 
-export type ChartReport = Omit<BaseReport, 'data'> & {
-  type: `${DashboardItemType.Chart}`;
+export type ChartReport = Omit<BaseReport, 'data' | 'type'> & {
+  type?: 'default'; // optional because of legacy reports
   data?: ChartData[];
 };
 
 export type DashboardItemReport = ViewReport | MatrixReport | ChartReport;
+
+export const isChartReport = (report?: DashboardItemReport | null): report is ChartReport => {
+  // If the report is a chart report, it will have a 'data' property or type 'default'
+  return report?.type === 'default' || !!(report && 'data' in report);
+};
+
+export const isMatrixReport = (report?: DashboardItemReport | null): report is MatrixReport => {
+  // If the report is a matrix report, it will have 'columns' and 'rows' properties or type 'matrix'
+  return report?.type === 'matrix' || !!(report && 'columns' in report && 'rows' in report);
+};
+
+export const isViewReport = (report?: DashboardItemReport | null): report is ViewReport => {
+  // If the report is a view report, it will have a 'data' property or type 'default' or 'rawDataExport'
+  return (
+    report?.type === 'default' || report?.type === 'rawDataExport' || !!(report && 'data' in report)
+  );
+};
```

Comment:
Is the reason we need to check anything other than the `report.type`so that we can support legacy reports?

Reply:
I think so!